FreeAPS/Sources/APS/Storage/OverrideStorage.swift

```diff
@@ -0,0 +1,330 @@
+import CoreData
+import Foundation
+import SwiftDate
+import Swinject
+
+/// Observer to register to be informed by a change in the current override
+protocol OverrideObserver {
+    func overrideDidUpdate(_ targets: [OverrideProfil?], current: OverrideProfil?)
+}
+
+protocol OverrideStorage {
+    func storeOverride(_ targets: [OverrideProfil])
+    func storeOverridePresets(_ targets: [OverrideProfil])
+    func presets() -> [OverrideProfil]
+    func syncDate() -> Date
+    func recent() -> [OverrideProfil?]
+    //  func nightscoutTretmentsNotUploaded() -> [NightscoutTreatment]
+    func current() -> OverrideProfil?
+    func cancelCurrentOverride() -> Decimal?
+    func applyOverridePreset(_ presetId: String) -> Date?
+    func deleteOverridePreset(_ presetId: String)
+}
+
+/// Class to manage the store of override and override preset
+final class BaseOverrideStorage: OverrideStorage, Injectable {
```

Comment:
change the try? to do /try / catch

Reply:
the enabled attribute is not a so good idea but it is not updated in function of the time (if the override is finished). That's why i don't use it. I don't delete it for "history" but could be deleted in the core data. What are the arguments to use is in regard of my proposal ?

Reply:
> the enabled attribute is not a so good idea because it is not updated in function of the time (if the override is finished). That's why i don't use it and propose to "calculate" when required by current() function. I don't delete it for "history" but could be deleted in the core data. What are the arguments to use is in regard of my proposal ?

My argument is that it basically creates to entities of data (in SQL terms, to tables) for storing virtually the same data, with only one property (in SQL terms, one column) being different. I'd suggest to update the `enabled` flag accordingly and get rid of the duplicate entity. 

I'm aware that this was not your doing and you are building on top of an existing two-entitiy-approach, but I am bringing it up because you are tackling a larger rewrite of Overrides and that includes both these entities. 

When the user cancels an override or it finishes, the respective entry needs to be updated to disabled. I can see how this causes you issues for chart display, so you could also consider adding an additional property `endDate` or simply calculating using `date` and `duration` (both already present). When the user cancels an override, you would have to update `duration` and `enabled`. When it runs out, only `enabled`. CoreData is a database, we can treat it like one and perform CRUD operations as we see fit :)

Reply:
Yes. a convincing argument 💪💪. I do the updates !

Reply:
@dnzxy , is this part of the conversation resolved now?